helix-core/src/main/java/org/apache/helix/api/rebalancer/constraint/dataprovider/PartitionQuotaProvider.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.helix.api.rebalancer.constraint.dataprovider;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * An interface for getting partition quota information.
+ */
+public interface PartitionQuotaProvider {
```

Comment:
if this is providing the weight of a partition, can we call it PartitionWeightProvider? Or, more generic, PartitionInfoProvider (and you can add other info if you need to)

Reply:
Should be PartitionWeightProvider based on previous comments and discussions.
Info would be too generic here, participant weight is strictly related to the participant capacity.

Given we have more date defined for the constraints, we can have more provider.
Constraint interface is not related to these providers. So we can always extend the provider list later.